sdk/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Util.hs

```diff
@@ -323,3 +324,30 @@ splitUnitId (unitIdString -> unitId) = fromMaybe (PackageName (T.pack unitId), N
     (name, ver) <- stripInfixEnd "-" unitId
     guard $ all (`elem` '.' : ['0' .. '9']) ver
     pure (PackageName (T.pack name), Just (PackageVersion (T.pack ver)))
+
+-- | Take a package version of regex "(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*" into
+-- a list of integers [Integer]
+splitPackageVersion :: PackageVersion -> Maybe [Integer]
+splitPackageVersion (PackageVersion raw) =
+  let pieces = T.split (== '.') raw
+  in
+  traverse (readMaybe . T.unpack) pieces
+
+data ComparePackageVersionError
+  = FirstVersionUnparseable PackageVersion
+  | SecondVersionUnparseable PackageVersion
+  deriving (Show, Eq, Ord)
+
+comparePackageVersion :: PackageVersion -> PackageVersion -> Either ComparePackageVersionError Ordering
```

Comment:
Here we don't use the `0.0.0` convention in other code because (I assume) packages don't adhere to that convention - is that true?